src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -119332,5 +119453,182 @@ class C<T>
+    static void FC<T>([DisallowNull]T x,    [AllowNull]ref T y) { y = x; }
+    static void FD<T>([DisallowNull]T x, [DisallowNull]ref T y) { y = x; }
+    static void FE<T>([DisallowNull]T x,    [MaybeNull]ref T y) { y = x; }
+    static void FF<T>([DisallowNull]T x,      [NotNull]ref T y) { y = x; }
+}";
+            var comp = CreateCompilation(new[] { source, AllowNullAttributeDefinition, DisallowNullAttributeDefinition, MaybeNullAttributeDefinition, NotNullAttributeDefinition });
+            comp.VerifyDiagnostics(
+                // (5,71): warning CS8601: Possible null reference assignment.
+                //     static void F1<T>(   [AllowNull]T x,               ref T y) { y = x; } // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "x").WithLocation(5, 71),
+                // (6,71): warning CS8601: Possible null reference assignment.
+                //     static void F2<T>(   [AllowNull]T x,    [AllowNull]ref T y) { y = x; } // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "x").WithLocation(6, 71),
+                // (7,71): warning CS8601: Possible null reference assignment.
+                //     static void F3<T>(   [AllowNull]T x, [DisallowNull]ref T y) { y = x; } // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "x").WithLocation(7, 71),
+                // (8,71): warning CS8601: Possible null reference assignment.
+                //     static void F4<T>(   [AllowNull]T x,    [MaybeNull]ref T y) { y = x; }
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "x").WithLocation(8, 71),
+                // (9,71): warning CS8601: Possible null reference assignment.
+                //     static void F5<T>(   [AllowNull]T x,      [NotNull]ref T y) { y = x; } // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "x").WithLocation(9, 71));
+        }
+
+        [Fact]
+        [WorkItem(38638, "https://github.com/dotnet/roslyn/issues/38638")]
+        public void MaybeNullT_29()
+        {
+            var source =
+@"#nullable enable
+class Program
+{
+    static T F1<T>()
+    {
+        (T t1, T t2) = (default, default);
+        return t1; // 1
+    }
+    static T F2<T>()
+    {
+        T t2;
+        (_, t2) = (default(T), default(T));
+        return t2; // 2
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (7,16): warning CS8603: Possible null reference return.
+                //         return t1; // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "t1").WithLocation(7, 16),
+                // (13,16): warning CS8603: Possible null reference return.
+                //         return t2; // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "t2").WithLocation(13, 16));
+        }
+
+        [Fact]
+        [WorkItem(38638, "https://github.com/dotnet/roslyn/issues/38638")]
+        public void MaybeNullT_30()
```

Comment:
What's the difference comparing to previous one?

Reply:
Good catch, thanks. Removed.

---
In reply to: [349946608](https://github.com/dotnet/roslyn/pull/39983#discussion_r349946608) [](ancestors = 349946608)